parsers/ZA.py

```diff
@@ -0,0 +1,962 @@
+                "QueryId": "",
+                "ApplicationContext": {
+                    "DatasetId": "41f7dfd2-71c9-4ad5-af19-cedd75514048",
+                    "Sources": [
+                        {
+                            "ReportId": "0d29e969-2444-4ce9-9e52-e3cc936474ca",
+                            "VisualId": "5a8d6dfdd561dd6a71a3",
+                        }
+                    ],
+                },
+            }
+        ],
+        "cancelQueries": [],
+        "modelId": 165350,
+    }
+    params = {"synchronous": "true"}
+    headers = {
+        "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36",
+        "Accept": "application/json, text/plain, */*",
+        "Accept-Language": "en-US,en;q=0.5",
+        "Activity-ID": "c3f74581-8bcd-cca5-7062-2cb380a00e8a",
+        "Request-ID": "18281c9a-2be5-ac40-418a-a22b288943a7",
+        "X-PowerBI-ResourceKey": "e4768909-4cc1-4164-a53a-4a5cab19ad76",
+        "Content-Type": "application/json; charset=utf-8",
+    }
+
+    response = post(
+        url=POWER_BI_URL,
+        json=data,
+        params=params,
+        headers=headers,
+    )
+
+    data = response.json()
+
+    only_power_values = data["results"][0]["result"]["data"]["dsr"]["DS"][0]["PH"][0][
+        "DM0"
+    ]
+
+    cleaned_power_bi_values = []
+
+    for i, value in enumerate(only_power_values):
+        cleaned_power_bi_values.append(value["C"])
+
+    assert len(cleaned_power_bi_values) > 0
+
+    return cleaned_power_bi_values
+
+
+def fetch_production(
+    zone_key: str="ZA",
+    session: Optional[Session]=None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> List[dict]:
+
+    if target_datetime is not None:
+        local_target_datetime = get(target_datetime).to(TIMEZONE)
+        local_one_week_ago = now(TIMEZONE).shift(days=-7)
+
+        if local_target_datetime < local_one_week_ago:
+            raise NotImplementedError(
+                f"No production data is available for {local_target_datetime}."
+            )
+
+    r = session or req_session()
```

Comment:
```suggestion
    r = session or Session()
```

Reply:
Done in commit 355a3467b0951490eb00d5d08ff2c31f6be1dd74